src/Platform.c

```diff
@@ -464,24 +464,157 @@ void PltWaitForConditionVariable(PLT_COND* cond, PLT_MUTEX* mutex) {
 #endif
 }
 
-uint64_t PltGetMillis(void) {
+//// Begin timing functions
+
+// These functions return a number of microseconds or milliseconds since an opaque start time.
+
+static bool has_monotonic_time = false;
+static bool ticks_started = false;
+
 #if defined(LC_WINDOWS)
-    return GetTickCount64();
-#elif defined(CLOCK_MONOTONIC) && !defined(NO_CLOCK_GETTIME)
-    struct timespec tv;
 
-    clock_gettime(CLOCK_MONOTONIC, &tv);
+static LARGE_INTEGER start_ticks;
+static LARGE_INTEGER ticks_per_second;
+
+void PltTicksInit(void) {
+    if (ticks_started) {
+        return;
+    }
+    ticks_started = true;
+    BOOL rc = QueryPerformanceFrequency(&ticks_per_second);
+    QueryPerformanceCounter(&start_ticks);
+}
+
+uint64_t PltGetMicroseconds(void) {
+    if (!ticks_started) {
+        PltTicksInit();
+    }
+    LARGE_INTEGER now;
+    BOOL rc = QueryPerformanceCounter(&now);
+    return (uint64_t)(((now.QuadPart - start_ticks.QuadPart) * 1000000) / ticks_per_second.QuadPart);
+}
+
+#elif defined(LC_DARWIN)
+
+static mach_timebase_info_data_t mach_base_info;
+static uint64_t start;
+
+void PltTicksInit(void) {
+    if (ticks_started) {
+        return;
+    }
+    ticks_started = true;
+    mach_timebase_info(&mach_base_info);
+    has_monotonic_time = true;
+    start = mach_absolute_time();
+}
+
+uint64_t PltGetMicroseconds(void) {
+    if (!ticks_started) {
+        PltTicksInit();
+    }
+    const uint64_t now = mach_absolute_time();
+    return (((now - start) * mach_base_info.numer) / mach_base_info.denom) / 1000;
```

Comment:
`mach_absolute_time()` is apparently discouraged now due to device fingerprinting concerns. It looks like `clock_gettime_nsec_np()` is the newer function they're recommending. It also avoids us having to deal with `mach_timebase_info()` too.

Reply:
Good catch, I had thought I looked at [these docs](https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time) which has a big yellow warning on it warning of App Store privacy-related things if you use it.  I will switch to the other one.

Reply:
Done, I also moved the `has_monotonic_time` variable into the final else block (Linux/Unix) since no other platform needs it. I didn't squash my commits this time but let me know if you would like the patch as 1 commit.